src/adapters/CKEditor5Adapter.ts

```diff
@@ -73,14 +80,36 @@ export class CKEditor5Adapter extends AbstractRichtextEditorAdapter {
     throw new Error('Unable to fetch editable element');
   }
 
-  replaceRanges(checkId: string, matchesWithReplacementArg: MatchWithReplacement[]) {
+  async replaceRanges(checkId: string, matchesWithReplacementArg: MatchWithReplacement[]) {
     if (!this.isInSourceEditingMode()) {
-      super.replaceRanges(checkId, matchesWithReplacementArg);
+      matchesWithReplacementArg.reverse();
+      for (const match of matchesWithReplacementArg) {
+        await this.replaceEachMatch(checkId, match);
+      }
     } else {
       window.alert('Action is not permitted in Source mode.');
     }
   }
 
+  async replaceEachMatch(checkId: string, match: MatchWithReplacement) {
+    await this.selectRanges(checkId, [match]);
+
+    const editor = this.getEditor();
+
+    await new Promise<void>((resolve) => {
+      editor.model.document.selection.once('change', (..._args) => {
+        resolve();
+      });
+    });
+
+    editor.model.change((writer) => {
+      const selectedRange = editor.model.document.selection.getFirstRange();
+      if (selectedRange) {
```

Comment:
Maybe we should check for selected range end > start to be sure selection is present.
The odds could be some other event got in between our calls which can possibly remove the selection we made

Reply:
what about inserting text at certain location right now we are not getting such matches but it's possible. In that case end == start.

Reply:
As discussed for insertion we should have other api then replace.